go_metrics/metrics/tests/test_graphite.py

```diff
@@ -25,13 +25,10 @@ def mk_graphite(self, handler=None):
         self.addCleanup(graphite.stop)
         returnValue(graphite)
 
-    @inlineCallbacks
     def mk_backend(self, **kw):
         kw.setdefault('persistent', False)
         backend = GraphiteBackend(kw)
-        backend.worker = yield self.get_worker({
-            'prefix': backend.config.prefix,
-            }, MetricWorker)
```

Comment:
Ah, right, that's we were doing this, forgot that we need a fake amqp worker. The issue is that `GraphiteBackend` already starts a worker, so the reactor is unclean at the end of each test because of the looping call started by the first worker if we just replace it with a different worker. I'll try tear down the original worker in `mk_backend` instead.